code/modules/explorer_drone/exodrone.dm

```diff
@@ -351,31 +352,36 @@ GLOBAL_LIST_EMPTY(exodrone_launchers)
 	. = ..()
 	GLOB.exodrone_launchers += src
 
-/obj/machinery/exodrone_launcher/attackby(obj/item/I, mob/living/user, params)
-	if(istype(I, /obj/item/fuel_pellet))
+/obj/machinery/exodrone_launcher/attackby(obj/item/weapon, mob/living/user, params)
+	if(istype(weapon, /obj/item/fuel_pellet))
 		if(fuel_canister)
-			to_chat(user, span_warning("There's already a fuel tank inside [src]!"))
+			to_chat(user, span_warning("There's already fuel loaded inside [src]!"))
 			return TRUE
-		if(!user.transferItemToLoc(I, src))
+		if(!user.transferItemToLoc(weapon, src))
 			return
-		fuel_canister = I
+		fuel_canister = weapon
 		update_icon()
 		return TRUE
-	else if(istype(I,/obj/item/exodrone) && user.transferItemToLoc(I, drop_location()))
+
+	if(istype(weapon, /obj/item/exodrone) && user.transferItemToLoc(weapon, drop_location()))
 		return TRUE
-	else
-		return ..()
 
-/obj/machinery/exodrone_launcher/crowbar_act(mob/living/user, obj/item/I)
-	. = ..()
-	if(fuel_canister)
-		to_chat(user, span_notice("You remove the fuel tank from [src]."))
-		fuel_canister.forceMove(drop_location())
-		fuel_canister = null
+	return ..()
+
+/obj/machinery/exodrone_launcher/crowbar_act(mob/living/user, obj/item/crowbar)
+	if(!fuel_canister)
+		return
+
+	to_chat(user, span_notice("You remove [fuel_canister] from [src]."))
+	fuel_canister.forceMove(drop_location())
+	fuel_canister = null
+	update_icon()
+	return TRUE
 
 /obj/machinery/exodrone_launcher/Destroy()
-	. = ..()
 	GLOB.exodrone_launchers -= src
+	fuel_canister = null
```

Comment:
Shouldn't it be qdel'd rather then nulled? or am I missing something

Reply:
I didn't qdel because machinery calls `dump_inventory_contents()` when destroyed, which spits the fuel out on the tile afterwards - I wasn't sure if it should be qdel'd or dropped after the launcher was destroyed

Reply:
Destroy() ought to be mostly side effectless, so like ahhh machines shouldn't drop their parts on Destroy, they should do it on deconstruct().
Should just be focused on removing something from the world. If you want it to break/drop things, that should get handled elsewhere

Reply:
Fine by me